crates/red_knot_python_semantic/resources/mdtest/call/methods.md

```diff
@@ -255,4 +255,81 @@ method_wrapper()
 method_wrapper(C(), C, "one too many")
 ```
 
+## `@classmethod`
+
+### Basic
+
+When a `@classmethod` attribute is accessed, it returns a bound method object, even when accessed on
+the class object itself:
+
+```py
+from __future__ import annotations
+
+class C:
+    @classmethod
+    def f(cls: type[C], x: int) -> str:
+        return "a"
+
+reveal_type(C.f)  # revealed: <bound method `f` of `Literal[C]`>
+reveal_type(C().f)  # revealed: <bound method `f` of `type[C]`>
```

Comment:
This could be `Literal[C]` instead of `type[C]`, but everything works anyway. This happens because we turn `Type::Instance("C")` into its meta type when a classmethod is called on an instance of `C`. This corresponds to the `type(obj)` call in this (pseudo) implementation:
```py
class classmethod:
    # …
    def __get__(self, obj, cls=None):
        if cls is None:
            cls = type(obj)
        return MethodType(self.f, cls)
```